File: src/main/java/mq/lombard/bottomsup/dao/BeerDao.java
package mq.lombard.bottomsup.dao;

import mq.lombard.bottomsup.thirdparty.Beer;
import mq.lombard.bottomsup.thirdparty.BeerApi;

import java.util.Set;

/** @author Pascal Lombard */
public class BeerDao {

  // Singleton
  private static BeerDao ourInstance = new BeerDao();

  private BeerDao() {}

  public static BeerDao getInstance() {
    return ourInstance;
  }

  public Beer getBeer(String name) {
    return BeerApi.get(name);
  }

  public Set<Beer> listBeers() {
    return BeerApi.list();
  }
}

File: src/main/java/mq/lombard/bottomsup/dao/BeerDao.kt
package mq.lombard.bottomsup.dao

import mq.lombard.bottomsup.thirdparty.Beer
import mq.lombard.bottomsup.thirdparty.BeerApi

/** @author Pascal Lombard
 */
object BeerDao {

  fun getBeer(name: String): Beer {
    return BeerApi.get(name)
  }

  fun listBeers(): Set<Beer> {
    return BeerApi.list()
  }
}
